src/middleware/thunk.js

```diff
@@ -1,10 +1,6 @@
-export default function thunkMiddleware(getState) {
-  return (next) => {
-    const recurse = (action) =>
-      typeof action === 'function' ?
-        action(recurse, getState) :
-        next(action);
-
-    return recurse;
-  };
+export default function thunkMiddleware({ dispatch, getState }) {
+  return (next) => (action) =>
+    typeof action === 'function' ?
+      action(dispatch, getState) :
```

Comment:
Looks like thunkMiddleware gives the call-site the ability to look at the current state of the app and dispatch actions based on that state. Did I get that right? When / how would we use this?